src/Entity/Contact.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Entity;
+
+use Ramsey\Uuid\Uuid;
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\ORM\Mapping as ORM;
+use Symfony\Component\Validator\Constraints as Assert;
+
+/**
+ * @ORM\Entity()
+ */
+class Contact
+{
+    /**
+     * @ORM\Id()
+     * @ORM\Column(type="guid")
+     */
+    private $id;
+
+    /**
+     * @Assert\NotBlank()
+     * @ORM\Column(type="string", length=63)
+     */
+    private $firstName;
+
+    /**
+     * @ORM\Column(type="string", length=63, nullable=true)
+     */
+    private $lastName;
+
+    /**
+     * @Assert\Email()
+     * @ORM\Column(type="string", length=127, nullable=true)
+     */
+    private $email;
+
+    /**
+     * @ORM\Column(type="string", length=31, nullable=true)
+     */
+    private $phoneNumber;
+
+    /**
+     * @ORM\ManyToMany(targetEntity="App\Entity\Address", cascade={"persist", "remove"})
+     */
+    private $addresses;
+
+    public function __construct()
+    {
+        $this->id = Uuid::uuid4()->toString();
+        $this->addresses = new ArrayCollection();
+    }
+
+    public function getId(): ?int
+    {
+        return $this->id;
+    }
+
+    public function getFirstName(): ?string
+    {
+        return $this->firstName;
+    }
+
+    public function setFirstName(string $firstName): self
```

Comment:
return void and remove `return $this`

Reply:
why ? this is setters chaining, as done in any entity

Reply:
Actually, it's what's wrongly done in most entities, as people tends to use anemic entitites.
With regard to software engineering practices and object oriented design, one should rather not use setters at all, but rather mutators (not linked to a specific property) and should ensure an object is always in a valid state (enforcing that using the constructor).

Reply:
Ok but I'm not sure it is something we should change in this PR. It will be a big change with a lot of effects as in Symfony, entities are supposed to have setters. Even in Doctrine, I'm not sure how to handle mutated object as entities are synced with an unit of work, keeping entity state, id...

Reply:
Oops didn't see those comments, Doctrine is not impacted by not creating getter and setter, for instance in my PR with the entity Event:
https://github.com/DarkmiraTour/community-event-manager/pull/146/files#diff-e6c235c76e1f8f021ad7da93a59059b6 
I have no issues about not adding setters, and for Symfony that doesn't impact either the entity, I know it has been made in several parts of the project, but the idea is to perfect what we do continuously, and put everything in the constructor can bring a lot of valid state of an object, as for the form update we use the Dto and create the update function taking a request Object as parameter and updating all fields in one call. 
With the implementation I made for the Event entity, I have no issue with Doctrine or Symfony, and I can process any CRUD actions without issue. And this will respect the object integrity as it's not possible to have the object created with no information.
If you are interested in this approach, we can discuss it directly on Slack with pleasure :)

Reply:
Those changes are addressed in issue #153 to be applied in the project globally